streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -93,6 +96,33 @@ long partitionTimestamp(final TopicPartition partition) {
         return queue.partitionTime();
     }
 
+    // creates queues for new partitions, removes old queues, saves cached records for previously assigned partitions
+    void updatePartitions(final Set<TopicPartition> newInputPartitions, final Function<TopicPartition, RecordQueue> recordQueueCreator) {
+        final Iterator<Map.Entry<TopicPartition, RecordQueue>> queuesIterator = partitionQueues.entrySet().iterator();
+        final Set<TopicPartition> removedPartitions = new HashSet<>();
+        streamTime = RecordQueue.UNKNOWN;
```

Comment:
What's the reasoning for resetting streamTime here?

Reply:
def from src: stream-time of this partition group defined as the largest timestamp seen across all partitions. E.g. we removed partition with largest value so we have to update stream-time with new value (max from rest of partitions) or if all of them removed it should be `UNKOWN`.
But I haven't investigate where this property is used so may be it's not very necessary.